packets.c

```diff
@@ -137,10 +137,14 @@ int change_tls_sni(const char *host, char *buffer, size_t bsize)
             || free_sz < diff) {
         return -1;
     }
-    *(uint16_t *)(sni + 2) = htons(old_sz + diff + 5);
-    *(uint16_t *)(sni + 4) = htons(old_sz + diff + 3);
-    *(uint16_t *)(sni + 7) = htons(old_sz + diff);
-    *(uint16_t *)(pad + 2) = htons(free_sz - diff);
+    uint16_t htons_sni2 = htons(old_sz + diff + 5);
+    uint16_t htons_sni4 = htons(old_sz + diff + 3);
+    uint16_t htons_sni7 = htons(old_sz + diff);
+    uint16_t htons_pad2 = htons(old_sz - diff);
```

Comment:
I'm not sure, but shouldn't it be `uint16_t htons_pad2 = htons(free_sz - diff);`?

Reply:
Yes, my mistake